machines-endpoints/pkg/machines-endpoints/main.go

```diff
@@ -411,11 +415,81 @@ func main() {
 		}
 	}
 
-	if *flgBMCConfigMap {
+	if *flgBMCReverseProxyConfigMap {
 		// create bmc-proxy configmap on all servers
 		err = client.updateBMCProxyConfigMap(ctx, machines)
 		if err != nil {
 			log.ErrorExit(err)
 		}
 	}
+
+	if *flgBMCLogCollectorConfigMap {
+		// create bmc-log-collector configmap of all servers that include boot servers
+		addersses, err := client.bmcListJson(machines)
+		if err != nil {
+			log.ErrorExit(err)
+		}
+		err = client.writeConfigMap(ctx, addersses)
+		if err != nil {
+			log.ErrorExit(err)
+		}
+	}
+}
+
+func (c client) bmcListJson(machines []Machine) (map[string]string, error) {
+	type MachineConfigMap struct {
+		Serial string `json:"serial"`
+		BmcIP  string `json:"bmc_ip"`
+		NodeIP string `json:"ipv4"`
+		Role   string `json:"role"`
+		State  string `json:"state"`
+	}
+	var ml []MachineConfigMap
+
+	for _, machine := range machines {
+		var m MachineConfigMap
+		if machine.Spec.BMC.IPv4 == "" {
+			continue
+		}
+		m.Serial = machine.Spec.Serial
+		m.BmcIP = machine.Spec.BMC.IPv4
+		m.NodeIP = machine.Spec.IPv4[0]
+		m.Role = machine.Spec.Role
+		m.State = machine.Status.State
+		ml = append(ml, m)
+	}
+
+	byteJSON, err := json.Marshal(ml)
+	if err != nil {
+		return nil, err
+	}
+	addresses := make(map[string]string)
+	addresses["serverlist.json"] = string(byteJSON)
```

Comment:
I don't think it is the job of this method to covert to `map[string][string]`.

Reply:
I don't understand what you mean, so please explain your intention in more detail.

Reply:
Updated the function name.